### PR TITLE
Move _to_datetime to util.py

### PR DESCRIPTION
**Pull request by twsearle**
# Move _to_datetime to util.py

Tidy up a little and move `drivers.gridded_forecast._to_datetime` and `drivers.ghrsstl4._to_datetime` to `util.py`.

### CODE REVIEW

**Comment by twsearle** [comment]
I am not super sure about this, instead of `import util.to_datetime` I wrote `from util import to_datetime as _to_datetime` this seems to fox `Mock` in the tests unless I import the function from the driver being tested. Sorry I am still learning testing!

File: test/test_util.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import pytest
+import cftime
+from datetime import datetime
+from unittest.mock import Mock, call, patch, sentinel
```
Comment (andrewgryan) [inline test/test_util.py]: Is this line needed?

Comment (twsearle) [reply test/test_util.py]: no we don't thanks I will remove

**Comment by twsearle** [comment]
Oops I think I still managed to miss some things after my rebase. I need to remove the references in load.py and navigate.py

**Comment by twsearle** [comment]
[test_navigator L134](https://github.com/twsearle/forest/blob/move_gridded_forecast_generic_functions/test/test_navigator.py#L134) is giving me errors, becuase we are no longer as forgiving about getting the driver name wrong.

**Comment by twsearle** [comment]
I broke `gridded_forecast.Navigator` in a couple of ways to test the test in `test_navigator` hopefully this is working OK now.

**Comment by andrewgryan** [comment]
@twsearle all looking good, as soon as the robots are happy I'll merge it

**Pull request merged**